install-and-configure/install/multi-cluster/federated-etl/aggregator.md

```diff
@@ -0,0 +1,139 @@
+
+## Configuring Aggregator
+
+### Prerequisites
+
+* Aggregator can only be configured in a Federated ETL environment
+* Must be using v1.107.0 of Kubecost or newer
+* Your *values.yaml* file must have set `kubecostDeployment.queryServiceReplicas` to its default value `0`. 
+
+### Tutorial
+
+Select from one of the two templates below and save the content as *aggregator.yaml*. This will be your configuration template required to set up Aggregator.
+
+Basic configuration:
+
+```
+kubecostAggregator:
+  replicas: 1
+  enabled: true
+  cloudCost:
+    enabled: true
+federatedETL:
+  federatedCluster: true
+kubecostModel:
+  containerStatsEnabled: true
+  federatedStorageConfigSecret: federated-store
+kubecostProductConfigs:
+  clusterName: YOUR_CLUSTER_NAME
+  cloudIntegrationSecret: cloud-integration
+  productKey:
+    enabled: true
+    key: YOUR_KEY
+prometheus:
+  server:
+    global:
+      external_labels:
+        cluster_id: YOUR_CLUSTER_NAME
+# when using managed identity/irsa, set the service account accordingly:
+serviceAccount:
+  create: false
+  name: kubecost-irsa-sa
+```
+
+Advanced configuration (for larger deployments):
+
+```
+kubecostAggregator:
+  replicas: 1
+  enabled: true
+  cloudCost:
+    enabled: true
+  env:
+    # governs parallelism of derivation step
+    # more threads speeds derivation, but requires significantly more 
+    # disk space for WAL storage
+    # default: 6
+    NUM_DERIVER_THREADS: 6
+    # log level
+    # default: info
+    LOG_LEVEL: debug
+  aggregatorDbStorage:
+    # governs storage size of aggregator DB storage
+    # !!NOTE!! disk performance is _critically important_ to aggregator performance
+    # ensure disk is specd high enough, and check for bottlenecks
+    # default: 128Gi
+    storageRequest: 128Gi
+  # tracing, starts embedded jaeger pod
+  # port forward to 16686 on the pod to access jaeger UI 
+  # provides instrumented traces used for troubleshooting performance issues
+  jaeger:
+    enabled: true
+    image: jaegertracing/all-in-one
+    imageVersion: latest
+federatedETL:
+  federatedCluster: true
+kubecostModel:
+  containerStatsEnabled: true
+  federatedStorageConfigSecret: federated-store
+kubecostProductConfigs:
+  clusterName: YOUR_CLUSTER_NAME
+  cloudIntegrationSecret: cloud-integration
+  productKey:
+    enabled: true
+    key: YOUR_KEY
+prometheus:
+  server:
+    global:
+      external_labels:
+        cluster_id: YOUR_CLUSTER_NAME
+# when using managed identity/irsa, set the service account accordingly:
+serviceAccount:
+  create: false
+  name: kubecost-irsa-sa
+```
+
+There is no baseline for what is considered a larger deployment, which will be dependent on load times in your Kubecost environment.
+
+Once you’ve configured your *aggregator.yaml*, create secrets using the following commands:
+
+```
+kubectl create secret generic cloud-integration -n kubecost --from-file=aggregator.yaml

```

Comment:
Not sure if this differs from the regular config, but I believe the from file should be `cloud-integration.json`.

```suggestion
kubectl create secret generic cloud-integration -n kubecost --from-file=cloud-integration.json
```

Reply:
The way this was explained is that they are creating the secret from the .yaml file they save in the step before this. Calling @ameijer to clarify this.

Reply:
For the aggregator secret, yes. Not for the cloud-integration secret.
https://docs.kubecost.com/install-and-configure/install/cloud-integration/multi-cloud#step-2-create-cloud-integration-secret